src/dcp/dcp.c

```diff
@@ -299,6 +312,293 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
+    rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey, 1, iod, &sgl, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "DAOS object fetch returned with errors: ", MFU_ERRF,
+	        MFU_ERRP(-MFU_ERR_DAOS));
+    }
+    return rc;
+}
+
+static int daos_copy_recx_array(daos_key_t *dkey,
+                           daos_key_t *akey,
+                           daos_handle_t *src_oh,
+                           daos_handle_t *dst_oh,
+                           daos_iod_t *iod)
+{
+    daos_anchor_t recx_anchor = {0}; 
+    int rc;
+    int i;
+    while (!daos_anchor_is_eof(&recx_anchor)) {
+        daos_epoch_range_t	eprs[5];
+        daos_recx_t		recxs[5];
+        daos_size_t		size;
+
+        /* list all recx for this dkey/akey */
+        uint32_t number = 5;
+        rc = daos_obj_list_recx(*src_oh, DAOS_TX_NONE, dkey, akey,
+	                        &size, &number, recxs, eprs,
+				&recx_anchor, true, NULL);
+
+        /* if no recx is returned for this dkey/akey move on */
+        if (number == 0) 
+            continue;
+
+        for (i = 0; i < number; i++) {
+            uint64_t    buf_len = recxs[i].rx_nr;
+            char        buf[buf_len];
+            d_sg_list_t sgl;
+            d_iov_t     iov;
+
+            /* set iod values */
+            (*iod).iod_type  = DAOS_IOD_ARRAY;
+            (*iod).iod_size  = 1;
+            (*iod).iod_nr    = 1;
+            (*iod).iod_recxs = &recxs[i];
+
+            /* set sgl values */
+            sgl.sg_nr     = 1;
+            sgl.sg_nr_out = 0;
+            sgl.sg_iovs   = &iov;
+
+            /* fetch recx values from source */
+            d_iov_set(&iov, buf, buf_len);	
+            rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+                                &sgl, NULL, NULL);
+
+            /* update fetched recx values and place in destination object */
+            rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+                                 &sgl, NULL);
+        }
+    }
+    return rc;
+}
+
+static int daos_copy_list_keys(daos_handle_t *src_oh,
+                          daos_handle_t *dst_oh)
+{
+    /* loop to enumerate dkeys */
+    daos_anchor_t dkey_anchor = {0}; 
+    int rc;
+    while (!daos_anchor_is_eof(&dkey_anchor)) {
+        d_sg_list_t     dkey_sgl;
+        d_iov_t         dkey_iov;
+        daos_key_desc_t dkey_kds[ENUM_DESC_NR]       = {0};
+        uint32_t        dkey_number                  = ENUM_DESC_NR;
+        char            dkey_enum_buf[ENUM_DESC_BUF] = {0};
+        char            dkey[ENUM_KEY_BUF]           = {0};
+
+        dkey_sgl.sg_nr     = 1;
+        dkey_sgl.sg_nr_out = 0;
+        dkey_sgl.sg_iovs   = &dkey_iov;
+
+        d_iov_set(&dkey_iov, dkey_enum_buf, ENUM_DESC_BUF);
+
+        /* get dkeys */
+        rc = daos_obj_list_dkey(*src_oh, DAOS_TX_NONE, &dkey_number, dkey_kds,
+                                &dkey_sgl, &dkey_anchor, NULL);
+        if (rc)
+            return -1;       
```

Comment:
@dsikich Should we log the `rc` here so there is an indication of what went wrong?